### PR TITLE
SDK-1207 - App version in node info API

### PR DESCRIPTION
**Pull request by ddrvar**
[Jira ticket](https://horizenlabs.atlassian.net/browse/SDK-1207?atlOrigin=eyJpIjoiYWExMDcyNjY4ZGI0NDNlODg1ZWY4ZDFkOGY1ZGI5NzQiLCJwIjoiaiJ9)

EonVersion is binded in `EvmAppModule` and added as an attribute to `AccountSidechainApp`. From there it is forwarded to `SidechainNodeApiRoute`. The default value is null, so for the UTXO model eonVersion won't be returned. Since eonVersion attribute is only added to `AccountSidechainApp` and not to `AbstractSidechainApp`, there is no need to update anything on the UTXO side.

Correct EonVersion is passed from the EON side, extracted from the package version in the same way as SDK version [here](https://github.com/HorizenOfficial/Sidechains-SDK/blob/7729cae9fd10670e57f02decc418b552906eab0c/sdk/src/main/scala/io/horizen/account/api/rpc/service/RpcUtils.scala#LL16C1-L17C1).

PR on the EON side is [here](https://github.com/HorizenOfficial/eon/pull/28).

### CODE REVIEW

File: examples/account/evmapp/src/main/java/io/horizen/examples/EvmAppModule.java [hunk 1]
```diff
@@ -104,5 +105,9 @@ public void configureApp() {
         bind(Integer.class)
                 .annotatedWith(Names.named("ConsensusSecondsInSlot"))
                 .toInstance(consensusSecondsInSlot);
+
+        bind(String.class)
+                .annotatedWith(Names.named("EonVersion"))
+                .toInstance(eonVersion);
```
Comment (MarcoOl94) [inline examples/account/evmapp/src/main/java/io/horizen/examples/EvmAppModule.java:111]: The SDK is a generic framework to build sidechains, here we are adding a reference to a specific project.
I think the best way here is to bind the value to a generic "AppVersion".
So i would rename:

- "EonVersion" -> "AppVersion"
- eonVersion -> appVersion

File: sdk/src/main/scala/io/horizen/account/AccountSidechainApp.scala [hunk 2]
```diff
@@ -55,7 +55,8 @@ class AccountSidechainApp @Inject()
    @Named("ApplicationStopper") applicationStopper: SidechainAppStopper,
    @Named("ForkConfiguration") forkConfigurator: ForkConfigurator,
    @Named("ChainInfo") chainInfo: ChainInfo,
-   @Named("ConsensusSecondsInSlot") secondsInSlot: Int
+   @Named("ConsensusSecondsInSlot") secondsInSlot: Int,
+   @Named("EonVersion") eonVersion: String
```
Comment (MarcoOl94) [inline sdk/src/main/scala/io/horizen/account/AccountSidechainApp.scala:59]: At this point that we generalized the name, can we use it also inside the UTXO model?

**Review by MarcoOl94 — CHANGES_REQUESTED** [review]

File: examples/account/evmapp/src/main/java/io/horizen/examples/EvmAppModule.java [hunk 3]
```diff
@@ -61,6 +61,7 @@ public void configureApp() {
 
         // It's integer parameter that defines slot duration. The minimum valid value is 10, the maximum is 300.
         int consensusSecondsInSlot = 12;
+        String eonVersion = "EON VERSION";
```
[the remark this answers is not in the document]
Comment (ddrvar) [reply examples/account/evmapp/src/main/java/io/horizen/examples/EvmAppModule.java]: Done
Empty string is accepted as binding value

File: sdk/src/main/scala/io/horizen/account/AccountAppModule.scala [hunk 4]
```diff
@@ -51,7 +51,8 @@ abstract class AccountAppModule extends com.google.inject.AbstractModule {
            @Named("CustomMessageProcessors") customMessageProcessors: JList[MessageProcessor],
            @Named("ApplicationStopper") applicationStopper : SidechainAppStopper,
            @Named("ForkConfiguration") forkConfigurator : ForkConfigurator,
-           @Named("ConsensusSecondsInSlot") secondsInSlot: Int
+           @Named("ConsensusSecondsInSlot") secondsInSlot: Int,
+           @Named("EonVersion") eonVersion: String
```
[the remark this answers is not in the document]
Comment (ddrvar) [reply sdk/src/main/scala/io/horizen/account/AccountAppModule.scala]: Done

File: sdk/src/main/scala/io/horizen/account/AccountSidechainApp.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (ddrvar) [reply sdk/src/main/scala/io/horizen/account/AccountSidechainApp.scala]: Done

**Review by MarcoOl94 — APPROVED** [review]

**Review by paolocappelletti — APPROVED** [review]

**Pull request merged**